trillian/ctfe/storage/storage.go

```diff
@@ -0,0 +1,17 @@
+package storage
+
+import (
+	"context"
+)
+
+// NonLeafCertificateChainStorage is an interface which allows CTFE binaries to use different storage implementations for non-leaf certificate chains.
+type NonLeafCertificateChainStorage interface {
+	// FindAll returns all key-value pairs of non-leaf certificate chains.
+	FindAll(ctx context.Context) (map[string][]byte, error)
+
+	// FindByHash returns the non-leaf certificate chain by the hash.
```

Comment:
s/by the/under, matching https://pkg.go.dev/github.com/hashicorp/golang-lru/v2#section-readme

Reply:
Updated to `// FindByHash returns the non-leaf certificate chain associated with the provided hash.` Matching the LRU cache does not align with other storage implementations, so I would prefer using a generic one.